dspace-server-webapp/src/main/java/org/dspace/app/rest/contentreports/Filter.java

```diff
@@ -0,0 +1,336 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.contentreports;
+
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiPredicate;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.app.rest.contentreports.ItemFilterUtil.BundleName;
+import org.dspace.authorize.factory.AuthorizeServiceFactory;
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.content.Bitstream;
+import org.dspace.content.Bundle;
+import org.dspace.content.Item;
+import org.dspace.core.Context;
+import org.dspace.services.factory.DSpaceServicesFactory;
+
+/**
+ * Available filters for the Filtered Collections and Filtered Items reports.
+ * In this enum, each item corresponds to a separate property, not values of
+ * a single property, hence the @JsonProperty applied to each of them.
+ * For each item, the annotation value is read through reflection and copied into
+ * the id property, which eliminates repetitions, hence reducing the risk or errors.
+ *
+ * @author Jean-François Morin (Université Laval)
+ */
+public enum Filter {
```

Comment:
It'd be nice to add tests for each of these filters...though they could just be part of the ITs for `ContentReportsRestController`, or similar.